cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/app/etl/batch/ETLMapReduce.java

```diff
@@ -299,6 +383,23 @@ public void destroy() {
     }
   }
 
+  private static class ErrorSink<IN, KEY_OUT, VAL_OUT> {
```

Comment:
this abstraction is probably unnecessary.  instead of a map from transform id to ErrorSink, we can just use a map of transform id to error dataset name, and just call context.write(errorDatasetName, AvroKey, NullWritable)